### DISCUSSION TITLE
Restrict a number of OTP credentials

### DISCUSSION THREAD

**Discussion by MosheHadadG** [keycloak/keycloak#26600] [Q&A] [answered]
Is there any way to restrict several OTP credentials (devices) for a user? The behavior I want to achieve is to allow only one OTP device to be active for a user.
The reason is that every time a user clicks on “forgot password” a new OTP credential will be added and is confused with three or more credentials.

**Comment by danielFesenmeyer** [comment, marked as answer]
It's possible to configure the "Reset OTP" authenticator. You can specify whether all, just one or no credentials are removed. This functionality has been implemented by @artur-baltabayev in this PR: https://github.com/keycloak/keycloak/pull/20572
Afaik, it's still possible for users to create more credentials (via Account UI), but at least there is less confusion introduced on reset. Maybe that's sufficient for your use case.

**Reply by MosheHadadG** [reply]
Great! thank you and for @artur-baltabayev
